packages/foundry/src/vault/strategy/autocompouding/liquidityPair/protocols/sushi/MiniChefV2Compounder.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: GPL-3.0
+// Docgen-SOLC: 0.8.15
+
+pragma solidity ^0.8.15;
+import { LiquidityPairBase, ERC20 } from "../../LiquidityPairBase.sol";
+import { IUniswapRouterV2 } from "../../../../../../interfaces/external/uni/IUniswapRouterV2.sol";
+import { IUniswapV2Pair } from "../../../../../../interfaces/external/uni/IUniswapV2Pair.sol";
+import { IMiniChefV2 } from "../../../../../../interfaces/external/sushi/IMiniChefV2.sol";
+import { IRewarder } from "../../../../../../interfaces/external/IRewarder.sol";
+
+contract LiquidityPairCompounder is LiquidityPairBase {
+  constructor(
+    address _native,
+    address _lpPair,
+    address _swapRouter,
+    address _vault,
+    address _strategist,
+    address[] memory _protocolAddresses,
+    uint256[] memory _protocolUints,
+    address[][] memory _rewardsToNativeRoutes,
+    address[] memory _nativeToLp0Route,
+    address[] memory _nativeToLp1Route
+  ) public {
+    native = _native;
+    lpPair = _lpPair;
+    swapRouter = _swapRouter;
+
+    vault = _vault;
+    strategist = _strategist;
+
+    protocolAddresses = _protocolAddresses;
+    protocolUints = _protocolUints;
+
+    rewardsToNativeRoutes = _rewardsToNativeRoutes;
+
+    _setUp(rewardsToNativeRoutes, nativeToLp0Route, nativeToLp1Route);
+  }
+
+  /*//////////////////////////////////////////////////////////////
+                          SETUP
+    //////////////////////////////////////////////////////////////*/
+
+  // Give allowances for protocol deposit and rewardToken swaps.
+  function _giveAllowances() internal override {
+    address chef = protocolAddresses[0];
+
+    ERC20(lpPair).approve(chef, type(uint256).max);
+    ERC20(native).approve(swapRouter, type(uint256).max);
+    ERC20(lpToken0).approve(swapRouter, type(uint256).max);
+    ERC20(lpToken1).approve(swapRouter, type(uint256).max);
+  }
+
+  /*//////////////////////////////////////////////////////////////
+                          HARVEST LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+  // Claim rewards from underlying protocol.
+  function _claimRewards() internal override {
+    address chef = protocolAddresses[0];
+    uint256 pid = protocolUints[0];
+
+    IMiniChefV2(chef).harvest(pid, address(this));
```

Comment:
Here we can simply call IAdapter(address(this) claim(). since the adapter hides this complexity